autoprecompiles/src/lib.rs

```diff
@@ -484,6 +501,31 @@ pub fn build<A: Adapter>(
     Ok(apc)
 }
 
+/// Generate optimistic constraints for superblock jumps (doesn't enforce the starting PC).
+fn superblock_pc_constraints<A: Adapter>(
+    block: &BasicBlock<A::Instruction>,
+) -> Vec<
+    OptimisticConstraint<
+        <<A as Adapter>::ExecutionState as ExecutionState>::RegisterAddress,
+        <<A as Adapter>::ExecutionState as ExecutionState>::Value,
+    >,
+> {
+    let mut res = vec![];
+    for (insn, pc) in block.other_pcs.clone() {
+        let left = OptimisticExpression::Literal(OptimisticLiteral {
+            instr_idx: insn,
+            val: LocalOptimisticLiteral::Pc,
+        });
+        let Ok(pc_value) = <<A as Adapter>::ExecutionState as ExecutionState>::Value::try_from(pc)
```

Comment:
I see, without this TryFrom we need to make the whole thing generic over the pc type hmm. Maybe it's not too bad, just adds a generic on BasicBlock basically.

Reply:
Yeah, this took me a little while... went with the `TryFrom<u64>` as we're assuming that's the PC type during execution. Let me know if you think this should be done differently